src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp

```diff
@@ -501,15 +501,30 @@ bool XMC_Flasher::isXMCReady()
     return xmcReady;
 }
 
-bool XMC_Flasher::isBMCReady()
+bool XMC_Flasher::isBMCReady(std::shared_ptr<pcidev::pci_device> dev)
```

Comment:
This function does not participate in ownership of argument device.  It should be 'const pci_device* dev' and be called as isBMCReady(mDev.get());